pkg/trace/agent/agent.go

```diff
@@ -269,15 +266,15 @@ var _ api.StatsProcessor = (*Agent)(nil)
 
 // ProcessStats processes incoming client stats in from the given language lang.
 func (a *Agent) ProcessStats(in pb.ClientStatsPayload, lang string) {
-	if in.Env == "" {
-		in.Env = a.conf.DefaultEnv
+	env := in.Env
+	if env == "" {
+		env = a.conf.DefaultEnv
 	}
-	in.Env = traceutil.NormalizeTag(in.Env)
-	out := stats.Payload{
-		HostName: in.Hostname,
```

Comment:
Should we remove `Hostname` from the payload? If the client sends a hostname different than the one that the agent uses on the traces corresponding to these stats, there may be a discrepancy and a user-experience problem in the application.

Reply:
The hostname for traces can be set with the `_dd.hostname` tag, this makes it possible for a tracer to send both traces and stats using a different hostname than the agent one.

Reply:
Wouldn't the proper handling of hostname here be to add a tag `_dd.hostname` to the counts if `stats.Payload.Hostname` is different from the agent hostname?